joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -1390,11 +1412,8 @@ void JointTrajectoryController::preempt_active_goal()
 
 void JointTrajectoryController::set_hold_position()
 {
-  trajectory_msgs::msg::JointTrajectory empty_msg;
-  empty_msg.header.stamp = rclcpp::Time(0);
-
-  auto traj_msg = std::make_shared<trajectory_msgs::msg::JointTrajectory>(empty_msg);
-  add_new_trajectory_msg(traj_msg);
+  // TODO(c-rizz) Should I use writeFromNonRT?
```

Comment:
This depends from where is this called. If from a callback than, yes, we should use RealtimeBuffer to make sure things work as they should.

Reply:
Let's discuss this after the other changes if it is still necessary.

Reply:
I use the RT-safe `add_new_trajectory_msg` now, no need for this hold-variable.